discovery-provider/src/tasks/cache_user_balance.py

```diff
@@ -35,46 +72,45 @@
 #
 #     enqueued User Ids in Redis that are *not* ready to be refreshed yet are left in the queue
 #     for later.
-
-
 def refresh_user_ids(
-    redis, db, token_contract, delegate_manager_contract, staking_contract, eth_web3
+    redis: Redis,
+    db: SessionManager,
+    token_contract,
+    delegate_manager_contract,
+    staking_contract,
+    eth_web3,
 ):
-    # List users in Redis set, balances decoded as strings
-    redis_user_ids = redis.smembers(REDIS_PREFIX)
-    redis_user_ids = [int(user_id.decode()) for user_id in redis_user_ids]
-
-    if not redis_user_ids:
-        return
+    with db.scoped_session() as session:
+        lazy_refresh_user_ids = get_lazy_refresh_user_ids(redis, session)
+        immediate_refresh_user_ids = get_immediate_refresh_user_ids(redis)
 
-    logger.info(
-        f"cache_user_balance.py | Starting refresh with {len(redis_user_ids)} user_ids: {redis_user_ids}"
-    )
+        logger.info(
+            f"cache_user_balance.py | Starting refresh with {len(lazy_refresh_user_ids)} "
+            f"lazy refresh user_ids: {lazy_refresh_user_ids} and {len(immediate_refresh_user_ids)} "
+            f"immediate refresh user_ids: {immediate_refresh_user_ids}"
+        )
+        all_user_ids = lazy_refresh_user_ids + immediate_refresh_user_ids
+        user_ids = list(set(all_user_ids))
 
-    with db.scoped_session() as session:
-        query = (
-            (session.query(UserBalance))
-            .filter(UserBalance.user_id.in_(redis_user_ids))
-            .all()
+        existing_user_balances: List[UserBalance] = (
+            (session.query(UserBalance)).filter(UserBalance.user_id.in_(user_ids)).all()
         )
 
         # Balances from current user lookup may
         # not be present in the db, so make those
-        not_present_set = set(redis_user_ids) - {user.user_id for user in query}
+        not_present_set = set(all_user_ids) - {
```

Comment:
Feels like this current_user lookup, which we're now doing in two places, is more trouble than it's worth IMO. I'd be supportive of just losing this whole flow of adding the is_current user in the first place. Would simplify things a lot.